doc/development.md

```diff
@@ -0,0 +1,107 @@
+# Development
+
+## Environment setup
+
+Fork the project on https://github.com/oVirt/vdsm.
+
+Clone your fork:
+
+    sudo dnf install -y git
+    git@github.com:{user}/vdsm.git
+
+Enable oVirt packages for Fedora:
+
+    sudo dnf copr enable -y nsoffer/ioprocess-preview
+    sudo dnf copr enable -y nsoffer/ovirt-imageio-preview
+
+Enable
+[virt-preview](https://copr.fedorainfracloud.org/coprs/g/virtmaint-sig/virt-preview/)
+package to obtain latest qemu and libvirt versions (and update if needed):
+
+    sudo dnf copr enable @virtmaint-sig/virt-preview
+
+Install additional packages for Fedora, CentOS, and RHEL:
+
+    sudo dnf install -y `cat automation/check-patch.packages`
+
+Create virtual environment for vdsm:
+
+    python3 -m venv ~/.venv/vdsm
+    source ~/.venv/vdsm/bin/activate
+    pip install --upgrade pip
+    pip install -r docker/requirements.txt
+    deactivate
+
+## Running the tests
+
+Before running vdsm tests, activate the environment:
+
+    source ~/.venv/vdsm/bin/activate
+
+When done, you can deactivate the environment:
+
+    deactivate
+
+Create storage for the tests:
+
+    make storage
+
+Alternatively, storage tests can be completely setup and run by doing:
+
+    make tests-storage
+
+See the tests [README](../tests/README.md#running-the-tests)
+for more information.
```

Comment:
Adding the section about creating storage for tests and running storage tests is not related to moving the text from the main readme. It is best done in a separate commit.